storage/remote/read.go

```diff
@@ -57,6 +61,10 @@ type querier struct {
 	client     *Client
 }
 
+func (q *querier) NewManager() storage.SelectManager {
+	return storage.NewConcurrentSelectManager(defaultSelectConcurrencyLimit)
```

Comment:
Hmm, won't this end up still running concurrent queries against the tsdb?

For a given query their should never be more than one select executing against the local tsdb at a time, otherwise CPU usage is unpredictable.

Reply:
If I haven't missed anything, this won't end up running queries against the tsdb.
Each Querier implementation has its own implementation to manage selects. This querier will only send concurrent read requests.

Reply:
That's not very obvious from the code, how are all these managers being layered?

Reply:
Exactly, It's more of a question of how Queriers are layered, and it's a little bit difficult to track down the behaviour of all the Queriers since there are a couple of high-level queriers that combine and wrap other Queriers. 

Right now, when you wrap a Querier, outermost Querier's behaviour will be adapted to use. For example, when you use fanout storage it uses primary Querier's (tsdb) select manager. That being said, I'm still doing experiments and trying to figure out the details of how to combine and layer Queriers.